FULL-feat-createAccountbook-colorchange

JunYoung7 opened the pull request:
### 📕 Issue Number

Linked #160 
<br/>

### 📗 작업 내역

> 구현 내용 및 작업 했던 내역

- 가계부 생성 페이지 색상 변경 모달 연결
![image](https://user-images.githubusercontent.com/61405355/102192437-23501c00-3efe-11eb-86c6-cb996e318a0e.png)

- colorLabel 에 클릭시 체크 모양이 생길 수 있도록 구현
![image](https://user-images.githubusercontent.com/61405355/102192723-7de97800-3efe-11eb-87e1-27a7bccd5593.png)

   - backgroundColor에 따라 체크 모양의 색깔이 변할 수 있도록 구현(색이 묻히지 않게)

- theme에 colorList 추가

- utils에 colorList 안에서 랜덤으로 색상 반환 함수 구현

- 가계부 생성 페이지가 randomColor를 colorList안에서 가져오도록 변경

<br/>

### 📘 작업 유형

- 신규 기능 추가
- 리펙토링

<br/>

### 📝 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- theme에 colorList를 추가하였고, utils에 colorList 안에서 랜덤으로 색상 반환 함수 구현하였습니다.

<br/><br/>

JunYoung7 committed:
feat : 가계부 생성 페이지 색상 변경 모달 연결

- 색상 변경 모달 적용을 위한 modal store 사용

JunYoung7 committed:
feat : 색상 변경 버튼 클릭 시 모달 state 변경


# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
refactor : 가계부 생성 페이지 수정

- backgroundColor 변화를 막기 위해 컴포넌트 밖으로 코드 이동

JunYoung7 committed:
feat : 체크 모양 svg 추가


# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
feat : ColorLabel에 svg 추가 및 이벤트 등록

- OnClick이 props로 있을 시에만 한해서 svg의 display를 조절할 수 있도록 구현

JunYoung7 committed:
feat : themes colorList 추가

- 가계부 생성 페이지 색깔 목록을 한정시키기 위한 colorList 추가

JunYoung7 committed:
refactor : getRandomColorInList 사용

- 랜덤한 색깔을 받아오는게 아닌 colorList 안에서 한정적으로 랜덤으로 받도록 수정

JunYoung7 committed:
feat : getRandomColorInList util 추가

- colorList 안에서 random한 색깔을 가져올 수 있도록 구현

JunYoung7 committed:
refactor : 색깔 선택 모달에서 colorList를 받아와서 사용하도록 수정


# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

do02reen24 commented on client/src/components/atoms/div/ColorLabel/ColorLabel.tsx:
```diff
@@ -20,10 +22,41 @@ const Div = styled.div<Props>`
-const ColorLabel: React.FC<Props> = ({ ...args }: Props) => {
-  return <Div {...args} />;
+const ColorLabel: React.FC<Props> = ({ onClick, size, backgroundColor }: Props) => {
+  const [checked, setChecked] = useState('none');
```
string값이 아닌 boolean값(true, false)으로 관리하는게 어떨까요?

JunYoung7 replied:
display에 관련된 변수라 다음과 같이 선언했습니다. 변수명을 display에 관련된 state임을 명시적으로 알 수 있도록 수정하겠습니다 !

leejiwoo2002 commented on client/src/components/atoms/div/ColorLabel/ColorLabel.tsx:
```diff
@@ -20,10 +22,41 @@ const Div = styled.div<Props>`
+      size={size}
+      backgroundColor={backgroundColor}
+      onClick={() => {
+        if (onClick) {
```
이름이 헷갈릴수도 있지 않을까 싶습니다
onClick 말고 다른 이름으로 변경하는건 어떨까요?

JunYoung7 replied:
넵 반영하겠습니다 !

leejiwoo2002 commented on client/src/theme/colorList.ts:
```diff
@@ -0,0 +1,11 @@
+const colorList = [
```
```javascript
getColorList = (startColor) => {
 return   [[startColor, '#FFBB00', '#ABF200', '#1DDB16'],
  ['#00D8FF', '#0054FF', '#5F00FF', '#FF00DD'],
  ['#FFA7A7', '#FAED7D', '#B7F0B1', '#B5B2FF'],
  ['#000000', '#D5D5D5', '#008299', '#D4F4FA'],
  ['#BDBDBD', '#C4B73B', '#670000', '#FFB4B4'],
  ['#EAEAEA', '#D1B2FF', '#FAF4C0', '#D3FFFF'],
  ['#FFFF6C', '#14899D', '#4B0CA5', '#845E00'],
];

}
```

이렇게 함수화 시켰어도 좋을것 같습니다

JunYoung7 replied:
startColor를 넣는 부분은 util로 빼고 순수한 colorList만 남기게끔 구현하였습니다. colorList는 함수 형태로 return 하게끔 코드 수정하겠습니다!

Zigje9 commented on client/src/components/atoms/div/ColorLabel/ColorLabel.tsx:
```diff
@@ -20,10 +22,41 @@ const Div = styled.div<Props>`
+const ColorLabel: React.FC<Props> = ({ onClick, size, backgroundColor }: Props) => {
+  const [checked, setChecked] = useState('none');
+
+  const changeChecked = () => {
+    if (checked === 'none') {
+      setChecked('');
+    } else {
+      setChecked('none');
+    }
+  };
```
빈 스트링과, none 보다는 true false 또는 checked, unchecked 이런식으로 의미를 부여하면 가독성이 좋을 것 같아요~

JunYoung7 replied:
변수명을 display 관련 변수로 display 속성 값임을 명시적으로 나타낼 수 있도록 수정하겠습니다!

Zigje9 commented on client/src/components/atoms/div/ColorLabel/ColorLabel.tsx:
```diff
@@ -20,10 +22,41 @@ const Div = styled.div<Props>`
+    <Div
+      size={size}
+      backgroundColor={backgroundColor}
+      onClick={() => {
+        if (onClick) {
+          onClick(backgroundColor);
+          changeChecked();
+        }
+      }}
```
if 문이 없으면 어떻게 되나요?

JunYoung7 replied:
다른 곳에서 colorLabel을 사용할 수도 있기 때문에 onClick event가 전달된 경우에만 함수가 실행될 수 있도록 구현하였습니다.

Zigje9 commented on client/src/theme/colorList.ts:
```diff
@@ -0,0 +1,11 @@
+const colorList = [
+  ['#FF0000', '#FFBB00', '#ABF200', '#1DDB16'],
+  ['#00D8FF', '#0054FF', '#5F00FF', '#FF00DD'],
+  ['#FFA7A7', '#FAED7D', '#B7F0B1', '#B5B2FF'],
+  ['#000000', '#D5D5D5', '#008299', '#D4F4FA'],
+  ['#BDBDBD', '#C4B73B', '#670000', '#FFB4B4'],
+  ['#EAEAEA', '#D1B2FF', '#FAF4C0', '#D3FFFF'],
+  ['#FFFF6C', '#14899D', '#4B0CA5', '#845E00'],
+];
+
+export default colorList;
```
color list를 빼서 관리하는점 잘하신것 같아요~

Zigje9 reviewed (approved):
고생하셨습니다~

JunYoung7 committed:
refactor : colorList 함수로 반환하게끔 형태 수정


# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
refactor: getColorList 함수 사용 부분 수정

- colorList -> getColorList 변경으로 인한 코드 수정

JunYoung7 committed:
refactor : ColorLabel 수정

- onClick에서 buttonEvent로 이름 수정
- checked 변수가 SVG의 display 속성임을 나타내주기 위한 checkDisplay로 이름 변경
